subsys/bluetooth/host/l2cap.c

```diff
@@ -2566,6 +2610,101 @@ int bt_l2cap_ecred_chan_connect(struct bt_conn *conn,
 
 	return err;
 }
+
+#if defined(CONFIG_BT_TESTING)
+static struct bt_l2cap_le_chan *l2cap_find_pending_reconf(struct bt_conn *conn)
+{
+	struct bt_l2cap_chan *chan;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&conn->channels, chan, node) {
+		if (BT_L2CAP_LE_CHAN(chan)->pending_rx_mtu) {
+			return BT_L2CAP_LE_CHAN(chan);
+		}
+	}
+
+	return NULL;
+}
+
+int bt_l2cap_chan_reconfigure(struct bt_l2cap_chan **chan, uint16_t mtu)
+{
+	struct bt_l2cap_ecred_reconf_req *req;
+	struct bt_conn *conn = NULL;
+	struct bt_l2cap_le_chan *ch;
+	struct net_buf *buf;
+	uint8_t ident;
+	int i;
+
+	for (i = 0; i < L2CAP_ECRED_CHAN_MAX; i++) {
+		if (!chan[i]) {
+			break;
+		}
```

Comment:
So `chan` is an array of channels, right? 

This check assumes that the caller supplies an array of exactly `L2CAP_ECRED_CHAN_MAX` channels. Is that intentional?

Reply:
it is array of pointers to channels

array can be up to L2CAP_ECRED_CHAN_MAX but can be NULL terminated if less is provided (this basically mimics behavior of  bt_l2cap_ecred_chan_connect())

Reply:
So the array needs to be NULL terminated, but that is, as far as I can tell, not documented anywhere. To me this seems like an extremely error prone design. 

I agree it makes sense to have the same design for all the L2CAP API functions, but this definitely needs to be documented at the very least, but possibly it's simply better to add a `chan_count` argument. 

It is fine to add in a different PR if you'd like, but something needs to be done.

Reply:
@Thalley It's documented now. Shall we create an improvement ticket for switching from NULL-terminated array to size parameter, and resolve this conversation?